spec.bs

```diff
@@ -574,15 +574,15 @@ To <dfn>parse a constructor string</dfn> given a string |input|:
         <dt>"<a for="constructor string parser/state">`protocol`</a>"</dt>
         <dd>
           1. If the result of running [=is a protocol suffix=] given |parser| is true:
-            1. Run [=compute should treat as a standard URL=] given |parser|.
-              <p class="note allow-2119">We must eagerly compile the protocol component to determine if it matches any [=special schemes=].  If it does then we treat the URLPattern constructor string as a "standard URL".  The determines if the pathname defaults to a "`/`" and also whether we should look for the username, password, hostname, and port components.  Authority slashes may also cause us to look for these components as well.  Otherwise we treat this as a "cannot be a base URL" and go straight to the pathname component.
-            1. If |parser|'s [=constructor string parser/should treat as a standard URL=] is true, then set |parser|'s [=constructor string parser/result=]["{{URLPatternInit/pathname}}"] to "`/`".
+            1. Run [=compute protocol matches a special scheme flag=] given |parser|.
+              <p class="note allow-2119">We must eagerly compile the protocol component to determine if it matches any [=special schemes=].  If it does then certain special rules apply.  It determines if the pathname defaults to a "`/`" and also whether we should look for the username, password, hostname, and port components.  Authority slashes may also cause us to look for these components as well.  Otherwise we treat this as a "cannot be a base URL" and go straight to the pathname component.
```

Comment:
It would be good to fix the allow-2119 while you're here. E.g. must -> need to, should -> will. Right now it's pretty confusing that you only "should" do those things, which implies an implementation might not do so.

Reply:
I guess the URL spec just calls it "path" and "cannot-be-a-base-URL path".

Reply:
I kind of like "pathname with slashes", but "can-be-a-base-URL pathname" might be more in line with URL spec terminology.

Reply:
I went with just "pathname" and "cannot-be-a-base-URL pathname" like the URL spec.  WDYT?